### ISSUE TITLE
Uninstallation doesn't delete files

### ISSUE THREAD

**Issue by patrikhuber** [desktop/desktop#4248]
I just went to `Control Panel\Programs\Programs and Features` on my Windows 10 and uninstalled `GitHub` (the date there said something like 10 Nov 2017, I think I might have installed earlier, so this probably shows the date of the last auto-update). It was pretty instantaneous.

I found now that it looks like all files in `C:\Users\Me\AppData\Local\GitHub` are still there.
The `Git Shell` shortcut in `C:\Users\Patrik\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\GitHub, Inc` is also still there but if I open it, it wants to install itself again.
I don't have anything else with the name "git" in it in the `Programs and Features` list.

![image](https://user-images.githubusercontent.com/4967343/37458656-9ad1f332-283d-11e8-8c3e-a672881d2a7a.png)
All subdirectories are full of files too, looks like all is still there.

What is happening with the uninstaller?

The log `TheLog.txt` doesn't contain any entry from the uninstallation.

**Comment by steveward** [comment]
👋 @patrikhuber 

It looks like you uninstalled the classic version of GitHub Desktop (this repository is for the completely rewritten version built using Electron). The classic version uses Microsoft's ClickOnce installer, which does not clean up after itself.

Here's a full breakdown of files/directories left behind:

1. The `GitHub` and `Git Shell` shortcuts 
2. The `GitHub` folder in the Local and Roaming app data directories. These are located `%LocalAppData%\GitHub` and `%AppData%\GitHub`.

There will also still be some files left in the ClickOnce Application Cache. You can run the following command to clear the application cache. Note, this clears the cache for _all_ ClickOnce applications.

1. Hold down the `Windows` key and type `R`.
2. Type in `rundll32 %windir%\system32\dfshim.dll CleanOnlineAppCache` in the run dialog and hit `Enter`.

Running the command might not clear every file. If you want to be absolutely thorough, you can delete the ClickOnce Application folder. This will be the entire `%LocalAppData%\Apps\2.0` directory.

Lastly, there will be one registry key you'll need to delete:

    HKEY_CURRENT_USER\Software\Classes\github-windows

This is the protocol handler that makes the "Clone in Desktop" button on GitHub.com launch the application.

Hope that helps!

**Comment by patrikhuber** [comment]
@steveward Hey thank you very much pointing out all these locations! I think I got rid of all of it now.

> If you want to be absolutely thorough, you can delete the ClickOnce Application folder. This will be the entire %LocalAppData%\Apps\2.0 directory.

I think this wasn't a particularly good idea... I deleted that folder but got lots of "... X ... is in use ..." errors, so I checked. It turns out that there are Lenovo Service Bridge files in there... and things that are actually running on my machine. Half of it is still there but quite a few files were removed - and no idea which, and what else, besides the LSB... :\

**Comment by Rudxain** [comment]
I have the same problem with the Electron-based app. The path `C:\Users\[USER_NAME]\AppData\Local\GitHubDesktop` contains non-user data, like executables and DLLs, **even after uninstalling.** The path `C:\Users\[USER_NAME]\AppData\Roaming\GitHub Desktop` seems to contain only cache and temporary files (IDK if it contains useful settings, but I assume most of those files can be easily re-generated and replaced)

**Comment by axsms** [comment]
While trying to resolve an issue with the default installation location (#13017), I discovered the same as @Rudxain. 

Uninstalling and then reinstalling gives no options to choose the location of install and simply re-installs to the previous/same location. In hopes that a complete uninstall would help, I manually removed files, folders, shortcuts, and registry entries. But, alas, there is actually no option to change the install location at all during install (#13091). So, the issue I was trying to resolve cannot be resolved by uninstalling and installing again to a new location.
